lottie-compose/src/main/java/com/airbnb/lottie/compose/LottieAnimatable.kt

```diff
@@ -160,12 +161,19 @@ private class LottieAnimatableImpl : LottieAnimatable {
     override var iterations: Int by mutableStateOf(1)
         private set
 
+    override var reverseOnRepeat: Boolean by mutableStateOf(false)
+        private set
+
     override var clipSpec: LottieClipSpec? by mutableStateOf(null)
         private set
 
     override var speed: Float by mutableStateOf(1f)
         private set
 
+    private val speedValue: Float by derivedStateOf {
```

Comment:
Does this need to be a property or can it be derived as a local variable on line 291?

Reply:
Done - 71154c638ff8d8d7bdc1863cef5a90f042575c6c

The value must remain a `derivedStateOf` instance otherwise the animation misbehaves. If you think that adds too much additional complexity to `onFrame` I can revert the change.

Thanks for the review!

Reply:
If `speedValue` is changed to a local `val` which isn't updated on L306 then the animation misbehaves and looks like this...

https://user-images.githubusercontent.com/1331002/194068499-bea1e6cc-ee9c-4606-9692-0dd6bc731658.mov

If you would prefer a `var` which is set on L287 and reassigned within `onFrame` on L306 I can make that change, it just seemed to complicate the function IMO instead of reading from a `derivedStateOf` property which does the calculation for you.

Reply:
Do you know why it's not working when it's a local val?

Reply:
I see. Could you add some documentation for this property since `speed` and `speedValue` is ambiguous on its own. It's not much better but maybe `rawSpeed` or `actualSpeed` or `speedWithRepeatMode`?

Reply:
Good idea. I added some docs to the property to explain why & when a negative value can be returned. I opted for `frameSpeed` for the property since that's more descriptive of how it is used. I didn't want to change `speed` -> `rawSpeed` as that would effect that public API of the class (as where the new `frameSpeed` property is private).

08f69299c99a1ecd2e99a51e548941bcb00ee62c